js/ui/number_box/number_box.mask.js

```diff
@@ -426,20 +430,24 @@ var NumberBoxMask = NumberBoxBase.inherit({
         var newValue = -1 * ensureDefined(this._parsedValue, null);
 
         if(this._isValueInRange(newValue)) {
-            this.option("value", newValue);
+            this._parsedValue = newValue;
```

Comment:
Where clearing the "this._parsedValue" variable ? Is possible not using this variable?

Reply:
It is not possible because ve must save 2 values: real editor value and parsed text value. This values may be different before valueChangeEvent